packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -104,7 +104,7 @@ module.exports = function watchmanCrawl(
 
             clocks[root] = response.clock;
             response.files.forEach(fileData => {
-              const name = root + path.sep + fileData.name;
+              const name = path.normalize(path.join(root, fileData.name));
```

Comment:
Well, you known what, the thing is we don't really need to 'normalize' paths logically speaking. Watchman ensures that the paths don't contain any ".." for example, so that's unnecessary work to parse and reform the paths. It's really just a matter of slashes/backslashes. I'll just do a regex replacement.